crates/vm/levm/src/opcode_handlers/system.rs

```diff
@@ -527,15 +508,40 @@ impl VM {
             None => Self::calculate_create_address(deployer_address, deployer_account_info.nonce)?,
         };
 
-        // 3. Account has nonce or code.
+        // touch account
+        self.accrued_substate.touched_accounts.insert(new_address);
+
+        let new_depth = current_call_frame
+            .depth
+            .checked_add(1)
+            .ok_or(InternalError::ArithmeticOperationOverflow)?;
+        // SECOND: Validations that push 0 to the stack and return reserved_gas
+        // 1. Sender doesn't have enough balance to send value.
+        // 2. Depth limit has been reached
+        // 3. Sender nonce is max.
+        if deployer_account_info.balance < value_in_wei_to_send
+            || new_depth > 1024
+            || deployer_account_info.nonce == u64::MAX
+        {
+            // Return reserved gas
+            current_call_frame.gas_used = current_call_frame
+                .gas_used
+                .checked_sub(max_message_call_gas.into())
+                .ok_or(VMError::Internal(InternalError::GasOverflow))?;
+            // Push 0
+            current_call_frame.stack.push(CREATE_DEPLOYMENT_FAIL)?;
+            return Ok(OpcodeSuccess::Continue);
+        }
+
+        // THIRD: Validations that push 0 to the stack dont return reserved gas but increment nonce
```

Comment:
```suggestion
        // THIRD: Validations that push 0 to the stack without returning reserved gas but incrementing deployer's nonce
```
Just a suggestion, you can change it to whatever you want. I just find that the actual comment can be improved

Reply:
Done in [c8b3f08](https://github.com/lambdaclass/ethrex/pull/1532/commits/c8b3f083c98c2c7fc43b892be5242084d6b8833c)